### PR TITLE
Fix #6214: restore BigInteger scale guard for DecimalNode conversions

### LINKED ISSUE #6214: `DecimalNode` to `BigInteger` conversion no longer guarded by `StreamReadConstraints.validateBigIntegerScale()`
In 2.x, every `DecimalNode` → `BigInteger` conversion went through `BaseJsonNode._bigIntFromBigDec()`, which calls `StreamReadConstraints.defaults().validateBigIntegerScale(value.scale())` before `BigDecimal.toBigInteger()` (added for #3864). That guard exists because `toBigInteger()` on a `BigDecimal` with huge scale magnitude is extremely expensive (computes 10^|scale|), while the textual form is tiny — `1e2000000000` is 12 characters, so `maxNumberLength` doesn't help.

In 3.x the guard is gone: `DecimalNode._asBigIntegerValueUnchecked()` is a bare `_value.toBigInteger()` (dropped in the #4970 number-accessor rework). Streaming parsers are still guarded (`ParserBase.convertNumberToBigInteger()`), and `BigIntegerDeserializer` / `TokenBuffer` still validate, but the `JsonNode` accessors are not:

```java
ObjectMapper mapper = JsonMapper.builder().enable(DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS).build();
JsonNode n = mapper.readTree("1e2000000000");
n.bigIntegerValue(); // CPU/memory DoS, no StreamConstraintsException
```

Same via `TreeTraversingParser.getBigIntegerValue()` (which is also how tree-backed format parsers such as TOML expose numbers), and via `asBigInteger()` / `bigIntegerValueOpt()` etc. — all six accessors funnel through `_asBigIntegerValueUnchecked()`.

Fix: restore the check in `DecimalNode._asBigIntegerValueUnchecked()` (using `StreamReadConstraints.defaults()`, as 2.x did, since nodes don't carry the configured constraints). PR to follow.

### PR DESCRIPTION
**Pull request by pjfanning**
Fixes #6214.

2.x guarded every `DecimalNode` → `BigInteger` conversion in `BaseJsonNode._bigIntFromBigDec()` with `StreamReadConstraints.defaults().validateBigIntegerScale(scale)` before `BigDecimal.toBigInteger()` (from #3864) — a 12-char value like `1e2000000000` otherwise makes `toBigInteger()` a CPU/memory DoS, and `maxNumberLength` can't catch it. The guard was dropped in the #4970 number-accessor rework; streaming parsers (`ParserBase`), `BigIntegerDeserializer` and `TokenBuffer` still have theirs, but the `JsonNode` accessors (and therefore `TreeTraversingParser.getBigIntegerValue()`, which tree-backed format parsers such as TOML rely on) do not.

This restores it in `DecimalNode._asBigIntegerValueUnchecked()` — the single funnel for all six `bigIntegerValue()`/`bigIntegerValue(default)`/`bigIntegerValueOpt()`/`asBigInteger()`/`asBigInteger(default)`/`asBigIntegerOpt()` accessors. "Unchecked" in that method's contract refers to the NaN / fractional-part checks the public accessors do first, not to this guard. Uses `StreamReadConstraints.defaults()` as 2.x did, since nodes don't carry the configured constraints; in 3.x `StreamConstraintsException` is unchecked so no sneaky-throw is needed. `DoubleNode`/`FloatNode` have no scale and are unaffected.

Tests: `DecimalNodeBigIntegerScaleTest` — all six accessors on a huge negative scale; the coercing `asBigInteger*()` variants on a huge positive scale (the strict ones already reject it as fractional, cheaply); via `TreeTraversingParser`; via `treeToValue(..., BigInteger.class)`; and boundary cases within/at the 100k limit. Fails with 2 errors without the fix; the `node` test package (436 tests) passes with it.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: release-notes/VERSION [hunk 1]
```diff
@@ -7,6 +7,8 @@ Versions: 3.x (for earlier see VERSION-2.x)
 
 3.3.0 (not yet released)
 
+#6214: `DecimalNode` to `BigInteger` conversion no longer guarded by
```
Comment (cowtowncoder) [inline release-notes/VERSION]: Wrong placement.

**Comment by cowtowncoder** [comment]
I think this should go against 3.1.

**Comment by pjfanning** [comment]
> I think this should go against 3.1.

retarget to 3.1

**Review by cowtowncoder — APPROVED** [review]

**Comment by cowtowncoder** [comment]
WIll merge later today.

**Comment by cowtowncoder** [comment]
Uh oh. PR seems to be merging 3.2, 3.x changes due to re-base...

**Comment by cowtowncoder** [comment]
@pjfanning Fixed to correctly target 3.1, but need to work on this bit more.
